feat(core): remove dependency to intl API

ocombe opened the pull request:
## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[x] Feature
```

## What is the current behavior?
We need the intl API for i18n pipes

Issue Number: #10809, #9524, #7008, #9324, #7590, #6724, #3429, #17576, #17478, #17319, #17200, #16838, #16624, #16625, #16591, #14131, #12632, #11376, #11187


## What is the new behavior?
We don't need the intl API anymore, we extract and use data from CLDR instead.

### Date pipe:
- new formats (breaking change), see [comparative document](https://docs.google.com/spreadsheets/d/12iygt-_cakNP1VO7MV9g4lq9NsxVWG4tSfc98HpHb0k/edit?usp=sharing)
- new optional parameter: `timezone` that lets you specify the timezone for a date (it will shift the date values based on the locale timezone so that you see it as it was from this other timezone).
- new optional parameter: `locale` that lets you use a specific locale for this pipe (instead of using the default `LOCALE_ID`)
 
### Plural pipe:
- new optional parameter: `locale` that lets you use a specific locale for this pipe (instead of using the default `LOCALE_ID`)
 
### Decimal pipe:
- new optional parameter: `locale` that lets you use a specific locale for this pipe (instead of using the default `LOCALE_ID`)

### Percent pipe:
- new optional parameter: `locale` that lets you use a specific locale for this pipe (instead of using the default `LOCALE_ID`)
 
### Currency pipe:
- the parameter `symbolDisplay` is no longer a boolean (it still works with a boolean, but it prints a warning to let you know that it changed), it’s now taking the values `code`, `symbol`, or `symbol-narrow` (the default is still code) which gives you access to more options for to show your currencies (e.g. the canadian dollar with the code `CAD` has the symbol `CA$` and the symbol-narrow `$`).
- new optional parameter: `locale` that lets you use a specific locale for this pipe (instead of using the default `LOCALE_ID`)

## Does this PR introduce a breaking change?
```
[x] Yes
```

All i18n pipes have been updated to not require the intl API anymore. Some results might vary, see the document to update from v4 to v5 for more information.

## TODO
- write a document to update from v4 to v5
- decide if we make the old pipes opt-in or opt-out
- find out how to make a deprecated common module that doesn't fail the tests
- create an access point from @angular/common/i18n that lets you import the locale that you want (need @IgorMinar's help for that)

prashant-andani committed:
docs(aio): update punctuation mark added in TOOLS guide

SamVerschueren committed:
fix(router): terminal route in custom matcher

Mobiletainment committed:
docs: fix spelling of case variants in naming.md

unify variant forms of spelling letter cases (upper-case, uppercase, lower case)

alxhub committed:
feat(common): new HttpClient API

HttpClient is an evolution of the existing Angular HTTP API, which exists
alongside of it in a separate package, @angular/common/http. This structure
ensures that existing codebases can slowly migrate to the new API.

The new API improves significantly on the ergonomics and features of the legacy
API. A partial list of new features includes:

* Typed, synchronous response body access, including support for JSON body types
* JSON is an assumed default and no longer needs to be explicitly parsed
* Interceptors allow middleware logic to be inserted into the pipeline
* Immutable request/response objects
* Progress events for both request upload and response download
* Post-request verification & flush based testing framework

DzmitryShylovich committed:
fix(router): should run resolvers for the same route concurrently

Fixes #14279

DzmitryShylovich committed:
fix(router): should navigate to the same url when config changes

Closes #15535

royling committed:
fix(router): canDeactivate guards should run from bottom to top

Closes #15657.

avdg committed:
docs(aio): cleanup rollup-config script

jasonhodges committed:
docs(aio): animations typos fixed

swseverance committed:
docs: correct grammar in CONTRIBUTING.md

swseverance committed:
docs(aio): fix typo

alexeagle committed:
fix(core): add needed closure compiler warning suppression

tinayuangao committed:
ci(aio): upload aio payload size to firebase

ci(aio): Add timestamp and change data

gkalpak committed:
refactor(upgrade): move shareable functionality to `UpgradeHelper` class (#17971)

This functionality can be potentionally re-used by the dynamic version.

dherges committed:
fix(aio): Typo in Setup Anatomy documentation page

Solves #17076

gkalpak committed:
feat(upgrade): fix support for `directive.link` in upgraded components
(#17971)

Although, pre- and post-linking functions are correctly called during directive
linking, directives with `link.post` would throw an error. Interestingly, having
`link.pre` only or defining `link: fn` (which is an alias for `link.post: fn`)
would not throw.

This commit removes this check and allows directives with pre- and/or
post-linking functions to work.

gkalpak committed:
fix(upgrade): bring the dynamic version closer to the static one
(#17971)

This commit changes the dynamic version of ngUpgrade to use `UpgradeHelper`,
thus bringing its behavior (wrt upgraded components) much closer to
`upgrade/static`. Fixes/features include:

- Fix template compilation: Now takes place in the correct DOM context, instead
  of in a detached node (thus has access to required ancestors etc).
- Fix support for the `$onInit()` lifecycle hook.
- Fix single-slot transclusion (including optional transclusion and fallback
  content).
- Add support for multi-slot transclusion (inclusing optional slots and fallback
  content).
- Add support for binding required controllers to the directive's controller
  (and make the `require` behavior more consistent with AngularJS).
- Add support for pre-/post-linking functions.

(This also ports the fixes from #16627 to the dynamic version.)

Fixes #11044

tinayuangao committed:
ci(aio): Add payload size limit file

tinayuangao committed:
ci(aio): Also track umd.min.js file size

chuckjaz committed:
feat(compiler-cli): new compiler api and command-line using TypeScript transformers

alexeagle committed:
build: circleci workflows to run lint in parallel

matsko committed:
fix(animations): do not delay style() values before a stagger() runs

Closes #17412

tinayuangao committed:
ci(aio): rename limits file and address comments

tinayuangao committed:
ci(aio): updated limits

tinayuangao committed:
ci(aio): add back deploy-preview

tinayuangao committed:
ci(aio): debug

tinayuangao committed:
ci(aio): remove umd

tinayuangao committed:
ci(aio): fix test

gkalpak committed:
build(aio): upgrade preview server dependencies

gkalpak committed:
test(aio): fix preview server tests on Windows

gkalpak committed:
refactor(aio): simplify preview server build events

gkalpak committed:
test(aio): add missing unit test for preview server

gkalpak committed:
feat(aio): enable previews for any PR

This commit introduces the ability to show previews for PRs by any author. It works as follows:

- The build artifacts of all PRs are uploaded to the preview server.
- Automatically verified PRs (i.e. from trusted authors or having a specific label) are deployed and
  publicly accessible as usual.
- PRs that could not be automatically verified are stored for later use (after re-verification).
- A PR can be marked as "trusted" and make its preview publicly accessible by adding the GitHub
  label specified in the `AIO_TRUSTED_PR_LABEL` env var of the preview server.

At the moment, there is no automatic mechanism for notifying the preview server about changes to the
PR's verification status. The PR's "visibility" will be checked and updated every time a new build
is uploaded.

gkalpak committed:
ci(aio): deploy previews for all PRs

PRs that could not be automatically verified will not be publicly accessible,
until manually verified.

brandonroberts committed:
docs: Updated router guide content and examples for paramMap

and queryParamMap, tracing, and incidental improvements.
closes #16991 and #16259 which it also fixes.

gkalpak committed:
docs(aio): document preview server HTTP status codes

gkalpak committed:
refactor(aio): enable `-u` flag on preview server scripts

gkalpak committed:
refactor(aio): provide fallback values for secrets (useful during dev)

pkozlowski-opensource committed:
refactor(forms): remove usage of deprecated Renderer

pkozlowski-opensource committed:
refactor(router): remove usage of deprecated Renderer

gkalpak committed:
test(aio): add e2e tests for non-public previews

vicb committed:
test(compiler): add a test for components not part of any `NgModule`

alexeagle committed:
build: No longer need to bazel build twice

wardbell committed:
fix(aio): prefix `location.assign` with `window.`

No practical effect but clears the TS compiler warning.

gkalpak committed:
refactor(aio): clean up `aio-contributor` template and styles

gkalpak committed:
fix(aio): fix links on `/about` in Firefox

Fixes #17661

vicb committed:
feat(compiler): add support `::ng-deep`

- /deep/ is deprecated and being removed from Chrome
- >>> is semantically invalid in a stylesheet
- sass will no longer support either in any version of sass

-> use ::ng-deep in emulated shadow DOM mode

Because the deep combinator is deprecated in the CSS spec,
`/deep/`, `>>>` and `::ng-deep` are also deprecated in emulated shadow DOM mode
and will be removed in the future.

see https://www.chromestatus.com/features/6750456638341120

wardbell committed:
feat(aio): cross reference readme and docs styleguide

pkozlowski-opensource committed:
refactor: remove unused imports of the deprecated Renderer

tinayuangao committed:
ci(aio): address comments

chuckjaz committed:
refactor(compiler): add tracking comment for self-import detection

The underlying cause of #17386 needs to be investigated and as a
simple string compare should have been valid.

matsko committed:
test(animations): add proper tests to check multiple combinations of :enter/:leave animation queries

tinayuangao committed:
ci(aio): Change the firebase token

Foxandxss committed:
build(aio): boilerplate wont be removed by default now

matsko committed:
fix(animations): properly collect :enter nodes that exist within multi-level DOM trees

Closes #17632

IgorMinar committed:
build: upgrade yarn to 0.24.6

IgorMinar committed:
build(aio): upgrade to angular@4.2.4 and zone.js

IgorMinar committed:
ci: add angular.io content related groups to .pullapprove.yaml

IgorMinar committed:
fix(router): export missing UrlMatcher and UrlMatchResult types

Fixes #15140

gkalpak committed:
fix(aio): build `scripts-js` before creating a new docker image for the preview server

When creating a new docker image for the preview server, the TypeScript source
code in `scripts-js/` is not copied over. Instead only the generated JavaScript
core in `scripts-js/dist/` are. Because of that, it is necessary to have run
`yarn build` before running `docker build`, so that the new docker image
contains the latest changes in `scripts-js/`.

This was previously part of the `create-image.sh` script, but was accidentally
removed in 21d213dfc.

gkalpak committed:
fix(aio): clean up non-public previews

The previous clean-up code for PR directories on the preview server assumed that
all directories were named after the PR number. With the changes introduced
in #17640 it is possible to have PR directories that do not follow that naming
convention (e.g. "non-public" directories).

This PR ensures that both public and non-public directories are removed when
cleaning up.

gkalpak committed:
feat(aio): use shorter URLs for previews

Use the 7 first characters of the 40-chars long SHAs for shorter/cleaner URLs.
The collision probability is extremely low (since all SHAs are further
"namespaced" under the corresponding PR). In case of a collision, the second PR
will not be deployed, in order to avoid overwriting the original build.

(This is a design decision to keep the implementation simple. It can be changed
later if necessary.)

gkalpak committed:
feat(aio): implement a way to check and update a PR\'s preview visibility

Previously, `BuildCreator#changePrVisibility()` would throw an error if the PR's
visibility was already up-to-date or if the PR directory did not exist (e.g. was
removed). This method was only used from inside `BuildCreator#create()`, which
had already checked for the existence of the directories.

This commit renames `changePrVisibility()` to `updatePrVisibility()` and makes
it more "forgiving" (i.e. it will only throw if both public and non-public
directories exist). This allows it to be used on events that may or may not have
caused the PR's visibility to change (e.g. a GitHub webhook triggered whenever a
PR's labels change).

jasonaden committed:
ci: add npm postinstall back to the lint step so node_modules doesn't get out of date

matsko committed:
fix(animations): do not remove container nodes when children are queried by a parent animation

Closes #17746

gkalpak committed:
refactor(aio): move script to a more relevant directory

gkalpak committed:
refactor(aio): use dedicated `constants.ts` file for e2e-specific constants

gkalpak committed:
feat(aio): add API endpoint for notifying about PR updates

This commit adds an API endpoint for notifying the preview server about PR
updates (`/pr-updated`). According to the update, the preview server can take
several actions. Currently, it will only check and (if necessary) update the
PR's preview visibility (but more actions could be supported in the future).
The API can be used with an automatic trigger (e.g. a GitHub webhook) to
instantly update a PR's preview visibility when it changes.

Fixes #16526

gkalpak committed:
refactor(aio): unify error messages for invalid requests to upload-server

Previously, there was a distinction between GET requests to invalid URLs and all
other requests. This was mainly because the upload-server only accepts GET
requests, but that is not a hard limitation and may change in the future.

Thus, it makes sense to return a 404 response for requests to invalid URLs
regardless of the method used.

gkalpak committed:
fix(aio): do not include hidden content in window title

The window title is derived based on the current document's `<h1>` heading. Such
headings may contain hidden/non-visible content (e.g. textual name of font
ligatures: `<i class="material-icons">link</i>`) that should not be included in
the title.

This commit fixes this by using `innerText` (instead of `textContent`) to
extract the visible text from the `<h1>` heading. It will still fall back to
`textContent` on browsers that do not support `innerText` (e.g. Firefox 44).

Fixes #17732

mary-poppins commented:
You can preview 64a274f2adecc5db2f6e68dfd8065f7f1bd14eb6 at https://pr17816-64a274f2adecc5db2f6e68dfd8065f7f1bd14eb6.ngbuilds.io/.

mary-poppins commented:
You can preview f3831bb488000aacd83c860f15eef2bad3746c1b at https://pr17816-f3831bb488000aacd83c860f15eef2bad3746c1b.ngbuilds.io/.

mary-poppins commented:
You can preview 955ce3c992f6bdcb482ec2aa2b3a682f8a9fcfb1 at https://pr17816-955ce3c992f6bdcb482ec2aa2b3a682f8a9fcfb1.ngbuilds.io/.

wardbell committed:
fix(aio): restore visible sidenav keyboard focus; shift TOC up

closes #17665
Restores keyboard focus that was removed by commit b8b91d3.
Raises the right-TOC by 20px (96px->76px) because was too far down.

To prevent keyboard focus on hidden child nodes,
also collapses inner expanded nodes when parent node is collapsed.
The implicit parent node of top nodes is always expanded.

petebacondarwin committed:
build(aio): refactor filterMemberDocs to be more general

Now it also filters function-overloads, hence the name change.

petebacondarwin committed:
build(aio): fix test description

mary-poppins commented:
You can preview c4171783d51b8ed8cb0f05b5fa8abf66d0fef675 at https://pr17816-c4171783d51b8ed8cb0f05b5fa8abf66d0fef675.ngbuilds.io/.

mary-poppins commented:
You can preview 0d10d6920f7f04f1f2103359730b5188155a016f at https://pr17816-0d10d6920f7f04f1f2103359730b5188155a016f.ngbuilds.io/.

StephenFluin committed:
docs(aio): Add resources contributor info

petebacondarwin committed:
build(aio): fix mergeDecoratorDocs processor

jasonaden committed:
ci: remove print-logs.sh to try to fix failing builds on Travis

petebacondarwin committed:
build(aio): fix matchUpDirectiveDecorators processor

matsko committed:
fix(animations): properly cleanup query artificats when animation construction fails

petebacondarwin committed:
build(aio): update doc-gen templates

matsko committed:
release: cut the 4.3.0-beta.1 release

matsko committed:
docs: add changelog for 4.3.0-beta.1

matsko committed:
fix(animations): properly handle cancelled animation style application

matsko committed:
docs: add changelog for 4.2.5

gkalpak committed:
ci(aio): avoid printing too large PWA report (and restore `print-logs.sh`)

There have been some issues lately with Travis jobs failing due to
`print-log.sh`. This is likely due to trying to print the Lighthouse PWA report,
which is too large.
This commit stops printing that report (since it was rarely used and is pretty
easy to acquire when needed) and restores the `print-logs.sh` script (that was
temporarily removed with dfcca66fd).

apetrosian committed:
docs(aio): fix quickstart-cli bad HTML escape chars

gkalpak committed:
ci(aio): use correct URL for preview in PWA score test

In 4268c8289, the preview URLs were changed to not use the whole SHA, but just
the 7 first characters.

mary-poppins commented:
You can preview f770024 at https://pr17816-f770024.ngbuilds.io/.

Foxandxss committed:
docs(aio): update quickstart to latest cli

gkalpak committed:
ci(aio): use valid database path for Firebase payload size upload

Firebase does not allow `.` in the path, so when trying to upload payload size
data for branches like `4.2.x`, the following error is thrown:

```
HTTP Error: 400, Invalid path: Invalid token in path
```

This commit fixes it by replacing `.` with `_` in branch names.

tinayuangao committed:
ci(aio): exclude changes in aio/content folder

tinayuangao committed:
ci(aio): Fix the payload script only check for changes in aio/scripts

alxhub committed:
refactor: move DOCUMENT from platform/browser to common

apetrosian committed:
docs(aio): fix visual-studio-2015 typo

This fixes #17283

jasonaden committed:
perf(core): refactor NgZone, decrease size by 1.2Kb (#17773)

- Remove getters
- Hide private methods for better property renaming

```
497893 May 31 11:26 core.umd.js
718073 May 31 11:26 core.umd.js.map
217108 May 31 11:26 core.umd.min.js
575092 May 31 11:26 core.umd.min.js.map
```

```
495594 May 31 11:28 core.umd.js
716943 May 31 11:28 core.umd.js.map
215826 May 31 11:28 core.umd.min.js
574401 May 31 11:28 core.umd.min.js.map
```
diff: 1,282

jasonaden committed:
feat(router): add router-level events for GuardsCheck and Resolve (#17601)

petebacondarwin committed:
build(aio): upgrade to dgeni-packages v0.20.0-rc.1

petebacondarwin committed:
build(aio): revert to general purpose search algorithm

Now that we have upgraded to the latest lunr search engine, the results
from the standard `search` method are more appropriate.
So we do not need to create our own special queries to get good results.

petebacondarwin committed:
build(aio): capture all the headings from a doc in the vFile.headings property

petebacondarwin committed:
build(aio): add terms from heading to the search index

gkalpak committed:
fix(upgrade): fix transclusion on upgraded components (#17971)

Previously, only simple, single-slot transclusion worked on upgraded components.
This commit fixes/adds support for the following:

- Multi-slot transclusion.
- Using fallback content when no transclusion content is provided.
- Destroy unused scope (when using fallback content).

Fixes #13271

vicb committed:
fix(compiler): fix types

vicb committed:
feat(compiler): adds support for quoted object keys in the parser

shefalisinha committed:
docs: shefali edits to Authors Style Guide (#17853)

wardbell committed:
docs: README - use correctly encoded link for npm badge (#17707)

closes #14990, an old PR that was approved but to old/difficult to merge

joejordanbrown committed:
docs(changelog): Correct typos (#16966)

wKoza committed:
docs: add animations package to CONTRIBUTING.md (#15413)

vicb committed:
fix(compiler): emits quoted keys only iff they are quoted in the original template

fixes #14292

vicb committed:
ci: add Chuck to the owners of core (#17892)

sjtrimble committed:
fix(aio): remove title from callout

petebacondarwin committed:
build(aio): abort doc-gen if an example is missing

Closes #16936

petebacondarwin committed:
build(aio): use dedicated overview for decorator API docs

petebacondarwin committed:
docs(aio): fix potential invalid jsdoc tag

petebacondarwin committed:
build(aio): fix mergeDecoratorDocs processor

petebacondarwin committed:
build(aio): fix matchUpDirectiveDecorators processor

petebacondarwin committed:
build(aio): abort doc-gen on dangling links

petebacondarwin committed:
ci(aio): raise polyfill payload limits

kapunahelewong committed:
docs(aio): add language service doc

vicb committed:
docs(aio): Update `NgStyle` and `NgClass` to use quoted keys

matsko committed:
feat(animations): support disabling animations for sub elements

Closes #16483

matsko committed:
fix(animations): ensure `:animating` queries collect previous animation elements properly

Commit:
feat(upgrade): propagate touched state of NgModelController

IgorMinar committed:
build: disable sauce connect logging (#17947)

this change is expected to mitigate the flakes on CI that occur when
we cat the log in the print-logs.sh

gkalpak committed:
docs(aio): add DevExtreme to the UI Components section of resources.json (#17939)

wardbell committed:
docs(aio): add instructions for example e2e tests to README (#17819)

matsko committed:
ci: allow chuck to approve animations (#17940)

Foxandxss committed:
feat(aio): serve-and-sync command (#17850)

adeora committed:
refactor(compiler-cli): remove the dependency on fs in codegen.ts (#17738)

vicb committed:
feat(compiler): update the schema by extracting from latest chrome (#17858)

adds:
- the slot element,
- the slot attribute on [Element],
- a few other attributes.

vicb committed:
fix(core): fix re-insertions in the iterable differ (#17891)

fixes #17852

kapunahelewong committed:
docs(aio): add Chuck's comments

jasonaden committed:
fix(router): encode URLs the same way AngularJS did (closer to spec) (#17890)

fixes #16067

kapunahelewong committed:
docs(aio): add npm install to Sublime instructions

matsko committed:
fix(animations): properly detect state transition changes for object literals

gkalpak committed:
fix(aio): fix text blurring when hovering over item in resources

Fixes #17907

gkalpak committed:
build(aio): remove unused script and GitHub token

Since 808bd4af4, we are no longer pre-verifying PRs before uploading the build
artifacts to the preview server, thus we no longer need the
`travis-preverify-pr.sh` script or the `GITHUB_TEAM_MEMBERSHIP_CHECK_KEY`
variable.

chuckjaz committed:
fix(language-service): do not crash when hovering over a label definitions (#17974)

Fixes: #17972

chuckjaz committed:
fix(tsc-wrapped): emit exports metadata in flat modules (#17893)

Fixes: #17888

chuckjaz committed:
fix(language-service): ignore hover of symbols not in the TypeScript program (#17969)

Fixes: #17965

gkalpak committed:
refactor(common): replace `Object.assign` with the spread operator (#17982)

`Object.assign` is not available in all supported browsers and one had to
provide a polyfill. This commit replaces `Object.assign` with the spread
operator (`...`), which TypeScript will transpile to ES5-compatible code.

matsko committed:
feat(animations): support :increment and :decrement transition aliases

adeora committed:
revert: "refactor(compiler-cli): remove the dependency on fs in codegen.ts (#17738)"

This reverts commit b1169014006e6be106c1481b9dbe644fa224d98d.

sjtrimble committed:
fix(aio): home & marketing styles cleanup (#17926)

gkalpak committed:
fix(aio): activate ServiceWorker updates asap (#17699)

Previouly, whenever a new ServiceWorker update was detected the user was
prompted to update (with a notification). This turned out to be more distracting
than helpful. Also, one would get notifications on all open browser tabs/windows
and had to manually reload each one in order for the whole content (including
the app) to be updated.

This commit changes the update strategy as follows:
- Whenever a new update is detected, it is immediately activated (and all
  tabs/windows will be notified).
- Once an update is activated (regardless of whether the activation was
  initiated by the current tab/window or not), a flag will be set to do a
  "full page navigation" the next time the user navigates to a document.

Benefits:
- All tabs/windows are updated asap.
- The updates are applied authomatically, without the user's needing to do
  anything.
- The updates are applied in a way that:
  a. Ensures that the app and content versions are always compatible.
  b. Does not distract the user from their usual workflow.

NOTE:
The "full page navigation" may cause a flash (while the page is loading from
scratch), but this is expected to be minimal, since at that point almost all
necessary resources are cached by and served from the ServiceWorker.

Fixes #17539

IgorMinar committed:
ci: disable sauce-connect logging for realz (#17995)

follow up on #17947

gkalpak committed:
docs: update required Node/npm versions in DEVELOPER.md

gkalpak committed:
build: remove duplicate dependency on TypeScript

TypeScript was added as a dependency in 9a7f5d580. I assume the intention was to
upgrade the existing devDependency instead.

vicb commented on gulpfile.js:
i18n -> cldr ?

vicb commented on packages/common/src/i18n/available_locales.ts:
Is this generated ? (If so please add a comment)

vicb commented on packages/common/src/i18n/currencies.ts:
Should `'symbol'` be quoted ?
Only props accessed via `obj['propName']` should be quoted (`obj.propName` should not so that they can be minified).

If we expect user to retrieve this info then we should add an API `return CURRENCIES[currencyName].symbol`

vicb commented on packages/common/src/i18n/data/locale_af-NA.ts:
add comment into generated files so that nobody update them manually

vicb commented on packages/common/src/i18n/data/locale_af-NA.ts:
we should think about what need to be quoted or not.
An alternative to quoted keys is an API returning obj.name

vicb commented:
Would be nice to split this in multiple commits:
- code
- generated code

vicb commented on tools/gulp-tasks/i18n/cldr-data.js:
glob ?

ocombe commented on tools/gulp-tasks/i18n/cldr-data.js:
This file is a copy from cldrjs or node-cldr, I just updated the paths to make it work here

vicb commented on tools/gulp-tasks/i18n/cldr-data.js:
why reduce ?

is it expected that we return either an array of a module why not always an array ?

ocombe commented on tools/gulp-tasks/i18n/cldr-data.js:
This file is a copy from cldrjs or node-cldr, I just updated the paths to make it work here

vicb commented on tools/gulp-tasks/i18n/cldr-data.js:
arrow function ?

alxhub committed:
feat(common): two missing features in HttpClient (#17996)

- Add params to HttpRequest API
- Add optional description to testing APIs

vicb committed:
fix(compiler): remove i18n markup even if no translations (#17999)

Fixes #11042

chuckjaz committed:
feat(compiler-cli): add parameters to ngc main needed by bazel rules (#17885)

ocombe committed:
fix(compiler-cli): fix relative source paths on windows for extracted msg (#17915)

Fixes #16639

filipesilva committed:
feat(aio): use new ngo (#17977)

gkalpak committed:
ci(aio): only deploy latest commits to staging/production (#17988)

Fixes #17941

vicb committed:
refactor: fix typos (#18000)

jasonaden committed:
style(http): fix linting error in http/testing (#18002)

jasonaden committed:
docs: add changelog for 4.2.6

jasonaden committed:
release: cut the 4.3.0-rc.0 release

jasonaden committed:
docs: add changelog for 4.3.0-rc.0

Commit:
fix(aio): add new common/http package into API list

petebacondarwin committed:
docs(aio): fix dangling links

gkalpak committed:
fix(aio): fix tab animations on CodeTabsComponent

In our attempt to remove the material ripple effect from tab labels, we were
killing all `transform`-based animations on other `md-tab-group` elements, such
as animating the content when entering/leaving. (This wasn't an issue on Chrome,
because it didn't respect our `!important` flag.)
This commit fixes it by properly hiding the ripple effect (using a feature
introduced in angular/material2@e4789c7b8) and allowing other animations to
execute normally.

Fixes #17998

kapunahelewong committed:
docs(aio): add George's fixes

petebacondarwin committed:
ci: add aio/content to aio group

This commit also simplifies the rules for the aio content and marketing groups

gkalpak committed:
test(aio): add tests for CodeTabsComponent

gkalpak committed:
refactor(aio): remove unnecessary comments/imports, improve indentation

gkalpak committed:
refactor(upgrade): clean up some types

gkalpak committed:
feat(upgrade): support lazy-loading Angular module into AngularJS app

jrevillas committed:
docs: fix style in helper text (#18055)

meDavid committed:
docs: Fix code example in the ChangeDetectorRef docs (#18051)

Minor example fix: mark the 'ref' constructor parameter as a private property

CrazyPython committed:
docs: various minor formatting changes (#18046)

CrazyPython committed:
docs: does please -> does (#18044)

Fixes grammar issue.

CrazyPython committed:
docs: Unbreak README Formatting (#18043)

- Makes headers consistent
 - Removes extra `)`
 - Makes browserstack line on a separate line

oktav777 committed:
docs: fix CHANGELOG typo (#18026)

ocombe committed:
test: git ignore e2e test folders (#17984)

trotyl committed:
docs(aio): fix host usage in styleguide (#17932)

Commit:
docs: fix 404 link to SFEIR website (#17913)

Also, I moved SFEIR training description to the "onsite" section since SFEIR doesn't deliver online training.

devoto13 committed:
fix(router): fix outdated homepage url in NPM package (#17899)

vprasanth committed:
docs: add @angular/animations to list (#17488)

@angular/animations was missing from the SemVer supported API surface list, now added.

Commit:
docs: improve grammar and wording on CONTRIBUTING.md (#17855)

matsko committed:
fix(animations): make sure @.disabled works in non-animation components

Note 4.3 only!

Prior to this fix when [@.disabled] was used in a component that
contained zero animation code it wouldn't register properly because the
renderer associated with that component was not an animation renderer.
This patch ensures that it gets registered even when there are no
animations set.

matsko committed:
fix(animations): make sure @.disabled works in non-animation components

Note 4.3 only!

Prior to this fix when [@.disabled] was used in a component that
contained zero animation code it wouldn't register properly because the
renderer associated with that component was not an animation renderer.
This patch ensures that it gets registered even when there are no
animations set.

trotyl committed:
docs: update ECMAScript info (#17801)

alexeagle committed:
docs: document using Bazel (#17701)

elgalu committed:
docs(benchpress): fix android broken links (#17568)

zarahzachz committed:
docs: fix typos and grammar (#17818)

kapunahelewong committed:
docs(aio): change Angular Module to NgModule (#16964)

Commit:
docs(aio): Added resource link to amexio-ng-extensions [duplicate of #17785] (#17790)

IgorMinar committed:
docs: adjust the release schedule with the 4.3.0-rc.0 release date

ocombe committed:
fix(compiler-cli): don't generate empty <target/> when extracting xliff

Fixes #15754

petebacondarwin committed:
build(aio): truncate API overview parameters at one line

petebacondarwin committed:
build(aio): render grouped overloads

alxhub committed:
fix(common): document HttpClient, fixing a few other issues

petebacondarwin committed:
build(aio): truncate API overview parameters at one line

petebacondarwin committed:
build(aio): fix githubLinks line numbers

The API docs have changed. The line numbers are now found in
`startingLine` and `endingLine` properties rather than the `location`
property, which moved into the `fileInfo` property anyway.

petebacondarwin committed:
build(aio): improve API docs layout and styling

petebacondarwin committed:
build(aio): upgrade to dgeni-packages 0.20.0-rc.5

This fixes unwanted extra overloads in interfaces and abstract classes
This provides the overload grouping support and fixes API doc line numbers

petebacondarwin committed:
build(aio): simplify the primary overload's anchor

The primary link to a member should simply be its name, and not require
the parameter list.

alexeagle committed:
ci: fast-forward to master/HEAD on all circle jobs (#18101)

This problem was introduced by a bad merge.

bmarkov committed:
docs(aio): Updated resources to UI Components (#18067)

With https:// the site gives console errors, because it loads fonts with http://

juleskremer committed:
docs(aio): add Dart as a language option (#18080)

alxhub committed:
fix(common): don't guess Content-Type for FormData bodies (#18104)

Fixes #18096

alxhub committed:
fix(aio): add <hr> in between overloads

chuckjaz committed:
fix(tsc-wrapped): report errors for invalid ast forms (#17994)

Fixes: #17993

chuckjaz committed:
fix(tsc-wrapped): support as and class expressions (#16904)

alxhub committed:
docs: rewrite 'Server Communication' doc guide to use HttpClient (#18079)

IgorMinar committed:
revert: ci(aio): exclude changes in aio/content folder

This reverts commit 1bf7ba87a0275022877bc87e9a1615327726d25c.

The commit caused master builds to fail with:

```
$ scripts/payload.sh
error Command failed with exit code 1.
error Command failed with exit code 1.
```
https://travis-ci.org/angular/angular/jobs/253405260#L1024

chuckjaz committed:
feat(compiler): do not evaluate metadata expressions that can use references (#18001)

alxhub committed:
fix(common): expose reportProgress option on HttpClient API (#18083)

IgorMinar committed:
revert: revert: ci(aio): exclude changes in aio/content folder

This reverts commit 72143e80daa86290455bdcf0da27132cdab4b84f.

it seems that the previous build failure was a travis flake.

reverting the revert...

alxhub committed:
fix(aio): document XSRF protection in HttpClient

alxhub committed:
feat(common): on-by-default XSRF support in HttpClient (#18108)

Fixes #18100

IgorMinar committed:
build(aio): turn on debugging of the payload.sh script to debug ci failures

cexbrayat committed:
docs: fix type error in HttpClient sample

gkalpak committed:
fix(aio): remove unused `news.html` file

Although outdated and not used, the file would be picked up and showed in search
results.

petebacondarwin committed:
ci: add GK and PBD to aio content and marketing groups

alxhub committed:
fix(common): rename HttpXsrfModule to HttpClientXsrfModule

alxhub committed:
Revert "fix(animations): make sure @.disabled works in non-animation components"

This reverts commit 01a2688848759dfc60c478f7641ca05338241c59.

jelbourn committed:
docs(aio): update jelbourn photo

alxhub committed:
release: cut the 4.3.0 release

alxhub committed:
docs: add changelog for 4.3.0

petebacondarwin committed:
build(aio): render type parameters of API function exports

Fixes #18123

alxhub committed:
Revert "revert: revert: ci(aio): exclude changes in aio/content folder"

This reverts commit 3d85f72652afb84ef8f99dba282c5ea05e93d386.

Still causing repeated flakes on master.

OzzieOrca committed:
docs(http): Make name of injected HttpTestingController consistent

wKoza committed:
docs: fix HttpClient sample

Commit:
docs(aio): fix HttpClient's interceptor sample

wKoza committed:
docs: fix HttpClient logging's sample

petebacondarwin committed:
docs(aio): fix up broken links

petebacondarwin committed:
docs(aio): fix broken image sources

petebacondarwin committed:
build(aio): fail doc-gen if referenced images are missing

petebacondarwin committed:
docs(aio): fix cheatsheet layout for narrow screens

* Tell the app that this will have no Table of Contents, since we have no
h2 headings anyway.
* Remove all the `nbsp;` from the code since that doesn't help with layout
* Remove side padding from sidenav-content when screen is narrow
* Restyle the cheatsheet table when the screen is narrow

happycollision committed:
docs(http): fix "Expecting and answering requests" example mistake

Possibly overlooked testing documentation mistake fixed:
- `http.get('/data')` probably ought to be paired with `httpMock.expectOne('/data')` instead of `httpMock.expectOne('/data')`.

jesselpalmer committed:
docs: fix typo

petebacondarwin committed:
fix(aio): remove title attribute from `CodeExampleComponent`

This was causing browser to add an unwanted tooltip that appeared
when the user hovers over the code.

See #17524

Commit:
docs(http): fixed syntax error in AuthInterceptor example

matsko committed:
fix(animations): capture cancelled animation styles within grouped animations

Closes #17170

chuckjaz committed:
fix(compiler): ensure jit external id arguments names are unique

Fixes: #17558, #17378, #8676

matsko committed:
fix(animations): always camelcase style property names that contain auto styles

Closes #17938

wKoza committed:
fix(aio): add quote to module

jnizet committed:
fix(common): send flushed body as error instead of null

fix #18181

vikerman committed:
docs(platform-server): add doc string for PlatformOptions

gkalpak committed:
fix(aio): do not wrap `<code-tabs>` tab labels

Fixes #17751

Commit:
docs(aio): fix HttpClient setting new header sample

matsko committed:
fix(animations): do not crash animations if a nested component fires CD during CD

Closes #18193

alxhub committed:
fix(platform-server): provide XhrFactory for HttpClient

gkalpak committed:
fix(upgrade): allow accessing AngularJS injector from downgraded module

nsanitate committed:
docs(aio): fix typo in Router documentation

Fix title and link to RouteConfigLoadEnd documentation

alxhub committed:
docs: add changelog for 4.3.1

alxhub committed:
release: cut the 5.0.0-beta.0 release

alxhub committed:
docs: add changelog for 5.0.0-beta.0

vicb commented on packages/common/src/pipes/i18n_plural_pipe.ts:
why ?

vicb commented on tools/gulp-tasks/cldr/extract.js:
clrdJs for consistency ?

ocombe commented on packages/common/src/pipes/i18n_plural_pipe.ts:
you told me in another PR that we should refactor the _ that were only here for dart and that we don't need anymore, so when I encounter one on the code that I'm updating, I change it

ocombe committed:
build(common): extract i18n locale data from cldr

ocombe committed:
build(common): add generated i18n locale data files

ocombe committed:
feat(core): remove dependency to intl API

BREAKING CHANGE: all i18n pipes have been updated to not require the intl API anymore. Some results might vary, see the document to update from v4 to v5 for more information.

mary-poppins commented:
You can preview 3482c17 at https://pr17816-3482c17.ngbuilds.io/.

mary-poppins commented:
You can preview 1a170b5 at https://pr17816-1a170b5.ngbuilds.io/.

mary-poppins commented:
You can preview 76aea6b at https://pr17816-76aea6b.ngbuilds.io/.

tytskyi commented:
just to note: there is still mention of Intl polyfill above this section: https://pr17816-76aea6b.ngbuilds.io/guide/pipes#built-in-pipes

ocombe commented:
oh thanks @tytskyi I'll fix that

mary-poppins commented:
You can preview 72665a6 at https://pr17816-72665a6.ngbuilds.io/.

mary-poppins commented:
You can preview 3de36af at https://pr17816-3de36af.ngbuilds.io/.

vicb commented on tools/gulp-tasks/cldr.js:
nit: remove `'./'` from `'./cldr/cldr-data'`

vicb commented on tools/gulp-tasks/cldr/extract.js:
`packages/common/src/i18n/data` should be configurable

vicb commented on tools/gulp-tasks/cldr/extract.js:
`utf8` is the default (you can remove the options)

vicb commented on tools/gulp-tasks/cldr/extract.js:
Factorize - do not repeat the headert

vicb commented on tools/gulp-tasks/cldr/extract.js:
reduce -> forEach + join ?

vicb commented on packages/common/src/i18n/data/locale_af.ts:
would it make sense to specify standalone only if different ?

vicb commented on packages/common/src/i18n/data/locale_af-NA.ts:
do this need to be exported ?

vicb commented:
You need to update the message of the last commit to something actionable - current is vague does not help users.

Use:
```
BREAKING CHANGE:

*What* is broken (Date, ...) and *how* ? 

BEFORE:

  <some snippet of what was working before>


AFTER:

  <how to update the code to make it work>
```

vicb commented on packages/common/src/common.ts:
Do we really need a `DeprecatedCommonModule`, would `DEPRECATED_ICU_PIPES` we enough ? (to be imported in the app module)

vicb commented on packages/common/src/i18n/localization.ts:
null ? return type is wring then ?

vicb commented on packages/common/src/i18n/localization.ts:
`locale.toLocaleLowerCase()` should be outside the closure

vicb commented on packages/common/src/i18n/localization.ts:
return early to avoid nested ifs

vicb commented on packages/common/src/i18n/localization.ts:
no else

vicb commented on packages/common/src/pipes/date_pipe.ts:
wrong ?

vicb commented on packages/common/src/pipes/date_pipe.ts:
Also do we support TZ names ?

vicb commented on packages/common/src/pipes/date_pipe.ts:
what does `G..GGG` means ? 
- `G..GGG` or
- `G`, `GG` and `GGG` ?

vicb commented on packages/common/src/pipes/date_pipe.ts:
We should probably document that locale data need to be imported

vicb commented on packages/common/src/pipes/date_pipe.ts:
`const formats = localeDatum.dateTimeSettings.formats`, factorize

vicb commented on packages/common/src/pipes/date_pipe.ts:
We should move the date formatting logic out of this pipe

vicb commented on packages/common/src/pipes/date_pipe.ts:
Did you consider using something else than a regexp here ?

vicb commented on packages/common/src/pipes/date_pipe.ts:
why ?

vicb commented on packages/common/src/pipes/date_pipe.ts:
Add a short comment on each function to explain what it does

vicb commented on packages/common/src/pipes/date_pipe.ts:
TODO ?

vicb commented on packages/common/src/pipes/date_pipe.ts:
TODOs ?

vicb commented on packages/common/src/pipes/i18n_plural_pipe.ts:
yes revert - explained in PM

vicb commented on packages/common/src/pipes/number_pipe.ts:
Is this still todo ?

syntax is `TODO(name): ...`

ocombe commented on packages/common/src/i18n/data/locale_af.ts:
if we create & export multiple object, we could reference the other one when they are the same

vicb commented on packages/common/src/pipes/number_pipe.ts:
some comment: move nb formatting outside of the pipe

ocombe commented on packages/common/src/i18n/data/locale_af-NA.ts:
I think it's necessary for AOT compilation

vicb commented on packages/common/src/pipes/number_pipe.ts:
isZero = digits.every(d => !d);

vicb commented on packages/common/src/pipes/number_pipe.ts:
for ?

vicb commented on packages/common/src/pipes/number_pipe.ts:
for ?

ocombe commented on packages/common/src/common.ts:
We cannot import both old and new pipes at the same time, so we need a common module that doesn't include the new pipes. I'm not sure if there is an alternative to another module ?

vicb commented on packages/common/src/pipes/number_pipe.ts:
"the string to go in front of" -> "prefix for" ?

ocombe commented on packages/common/src/i18n/localization.ts:
no, the comment is wrong, it was returning null before but I changed it so that it throws an error instead and forgot to update the comment

vicb commented on packages/common/src/pipes/number_pipe.ts:
"three components that can be used for formatting the number." What is the added value of this part ?

ocombe commented on packages/common/src/pipes/date_pipe.ts:
it's "`G`, `GG` and `GGG`", that's how they write it on the cldr docs: http://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table

vicb commented on packages/common/src/pipes/number_pipe.ts:
remove the doc here and add it to where it belongs (ParsedNumber)

ocombe commented on packages/common/src/pipes/date_pipe.ts:
No, I've copied this from the AngularJS code. What's the alternative?

ocombe commented on packages/common/src/pipes/date_pipe.ts:
We don't support all formats offered by CLDR (http://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table), a lot of them seem kinda useless, or require too many data for us. I added this note to support them in case we decide to add "supplemental locale data".

vicb commented on packages/common/test/localization_spec.ts:
why about importing from the index that contain all the locales ?

ocombe commented on packages/common/src/pipes/number_pipe.ts:
Yes, we don't have a pipe for scientific format. I'm not sure if we need one but it could be useful I think.

vicb commented on packages/common/test/localization_spec.ts:
does not case:
- case sensitivity,
- parent locale

vicb commented on packages/core/src/i18n/ng_locale.ts:
why ?

ocombe commented on packages/common/src/pipes/number_pipe.ts:
so that we can support big integers I think (this code comes from angularjs)

vicb commented on packages/core/src/i18n/ng_locale.ts:
why ?

ocombe commented on packages/common/test/localization_spec.ts:
yes, I wasn't sure if we wanted to keep that index or not

ocombe commented on packages/core/src/i18n/ng_locale.ts:
to avoid ts errors when I was referencing using data[key], not sure if it's still needed, I'll check

vicb commented on packages/common/src/common.ts:
> We cannot import both old and new pipes at the same time

Well we can at least import the common module (with the new pipes) then a `DeprecatedPipeModule`.
There is not need to have a `DeprecatedCommon` for 3 pipes

vicb commented on packages/common/src/pipes/date_pipe.ts:
may be expand it here so that's it's clear

vicb commented on packages/common/src/pipes/date_pipe.ts:
switch / case

vicb commented on packages/common/src/pipes/date_pipe.ts:
There should be no TODO without comment - either add a description or remove it

vicb commented on packages/common/src/pipes/number_pipe.ts:
remove the todo, create an issue on GH.
What's `Scientific` below ?

vicb commented on packages/common/src/pipes/number_pipe.ts:
I was only referring to the sentence "three components that can be used for formatting the number." - useless IMO

vicb commented on packages/core/src/i18n/ng_locale.ts:
`data[key]` could cause issue with minification

ocombe commented on packages/common/src/pipes/number_pipe.ts:
a placeholder that isn't used yet, I'll remove it

ocombe commented on packages/common/src/pipes/date_pipe.ts:
true, I'll regroup all those comments and add a description

ocombe commented on packages/common/src/pipes/date_pipe.ts:
that would be a lot of code, and since we use a regexp in angularjs and in the old date pipe in angular as well, I suppose that it is more efficient

ocombe commented on packages/common/src/pipes/date_pipe.ts:
You're right, it's showing GMT+x not TZ name, and the syntax is different, I'll update those

ocombe commented on packages/common/src/pipes/date_pipe.ts:
it's used for hours with format `h` and `hh` (in AM/PM). if we have an hour of 0 and an offset of -12, then it's equivalent to "12" (12am = 00:00 = midnight)
